src/build/rollup-importmap-plugin.ts

```diff
@@ -0,0 +1,65 @@
+import { Plugin } from "rollup";
+import fs from "node:fs/promises";
+import path from "node:path";
+import { pathToFileURL } from "node:url";
+import { fetch } from "@jspm/generator";
+import { Flags } from "../types";
+import { getGenerator, JspmError } from "../utils";
+
+const isValidUrl = (url: string) => {
+  try {
+    return Boolean(new URL(url));
+  } catch (e) {
+    return false;
+  }
+};
+
+export const RollupImportmapPlugin = async (flags: Flags): Promise<Plugin> => {
+  /*
+    Install without a freeze might bump the versions.
+    We would like to maintian 1:1 on what users defined in importmap.
+  */
+  const generator = await getGenerator({ ...flags, freeze: true });
+  await generator.install();
+
+  return {
+    name: "rollup-importmap-plugin",
+    resolveId: async (id: string, importer: string) => {
+      if (isValidUrl(id)) {
+        const url = new URL(id);
+        if (url.protocol === "deno:" || url.protocol === "node:") {
+          return { id, external: true };
+        }
+      }
+
+      try {
+        const resolved = generator.importMap.resolve(id, importer);
+        return { id: resolved };
+      } catch (err) {
+        return { id, external: true };
```

Comment:
We probably do want to provide module not found errors. And then have some handling for externals. At the very least perhaps we should use warnings?

I'm happy for us to follow-up on this though and merge now if you prefer.

Reply:
Should we through a console error for now and then allow users to provide a option to install the deps automatically down the line ?

Adding a `console.warn` for now 👍